src/test/java/com/textmagic/sms/TextMagicMessageServiceTest.java

```diff
@@ -0,0 +1,492 @@
+package com.textmagic.sms;
+
+import org.junit.runner.RunWith;
+import org.junit.Before;
+import org.junit.Test;
+import org.jmock.integration.junit4.JUnit4Mockery;
+import org.jmock.integration.junit4.JMock;
+import org.jmock.Expectations;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.allOf;
+import org.hamcrest.Matchers;
+import com.textmagic.sms.core.invoker.HttpServiceInvoker;
+import com.textmagic.sms.core.invoker.ServiceInvokerException;
+import com.textmagic.sms.core.parsing.TextMagicResponseParser;
+import com.textmagic.sms.core.parsing.ResponseParsingException;
+import com.textmagic.sms.dto.*;
+import com.textmagic.sms.exception.ServiceTechnicalException;
+import com.textmagic.sms.exception.ServiceBackendException;
+import com.textmagic.sms.util.StringUtils;
+
+import java.util.*;
+import java.math.BigDecimal;
+
+import static junit.framework.Assert.*;
+
+/**
+ * Date: 23.05.2009
+ *
+ * @author: bagmanov
+ */
+@RunWith(JMock.class)
+public class TextMagicMessageServiceTest {
+    JUnit4Mockery context = new JUnit4Mockery();
+    TextMagicMessageService service;
+    private static final String LOGIN = "myLogin";
+    private static final String PASSWORD = "myPassword";
+    HttpServiceInvoker serviceInvoker;
+    TextMagicResponseParser responseParser;
+    private static final String MY_TEXT = "my text";
+    private static final String CORRECT_MSISDN = "79261234567";
+
+    @Before
+    public void setUp() throws Exception {
+        serviceInvoker = context.mock(HttpServiceInvoker.class);
+        responseParser = context.mock(TextMagicResponseParser.class);
+
+        service = new TextMagicMessageService(LOGIN, PASSWORD);
+        service.setInvoker(serviceInvoker);
+        service.setParser(responseParser);
+    }
+
+    @Test
+    public void testSend() throws Exception{
+        final List<SentMessage> messages = new ArrayList<SentMessage>();
+        SentMessage message =  new SentMessage();
+        message.setId(2L);
+        messages.add(message);
+        final String invokerResponse = "dummy response";
+        context.checking(new Expectations() {{
+            one(serviceInvoker).invoke(
+                    with(equalTo(LOGIN)),
+                    with(equalTo(PASSWORD)),
+                    with(equalTo("send")),
```

Comment:
The String literal 'send' appears 5 times in this file; the first occurrence is on line 63